### PR TITLE
Path builder

### PR DESCRIPTION
**Pull request by chrisktaylor**
## Features
- Added class/methods for file generation paths from config file and directory creation.

[Delivers# LASKG-2]

### CODE REVIEW

File: src/core/PathBuilder.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ * PHP version 5.6
+ *
+ * @category  GENERATOR
+ * @package   LASKGEN
+ * @copyright (c) 2015, The Fable Project
+ * @author    CKT <chrisktaylor@gmail.com>
+ * @license   LASKGEN APACHE LICENSE 
+ *            https://github.com/Fable-Project/lask-gen/blob/master/LICENSE
+ */
+namespace FableProject\LaskGen\Core;
```
Comment (GSumisu) [inline src/core/PathBuilder.php]: not your fault, it's mine.. but LaskGen is now Laskgen (my mistake), I'll change though don't worry about it, just letting you know for future work.

Comment (chrisktaylor) [inline src/core/PathBuilder.php]: If you're making the change, please make sure you change it in the unit test as well.

File: src/test/TestPathBuilder.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        
+        $this->assertFalse($returnedValue);
+    }
+    
+    /**
+     * Standalone test to ensure that the correct path is created.
+     */
+    public function testMakeCreatesCorrectPath()
+    {
+        $pathBuilder = new \FableProject\LaskGen\Core\PathBuilder($this->tree);
```
Comment (GSumisu) [inline src/test/TestPathBuilder.php]: changing namespace (comment above)

File: src/test/TestPathBuilder.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+ * PHP version 5.6
+ *
+ * @category  GENERATOR
+ * @package   LASKGEN
+ * @copyright (c) 2015, The Fable Project
+ * @author    CKT <chrisktaylor@gmail.com>
+ * @license   LASKGEN APACHE LICENSE 
+ *            https://github.com/Fable-Project/lask-gen/blob/master/LICENSE
+ */
+namespace FableProject\LaskGen\Test;
```
Comment (GSumisu) [inline src/test/TestPathBuilder.php]: changing namespace (comment above)

File: src/test/TestPathBuilder.php [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        ];
+    
+    /**
+     * Test that the find method returns the correct string for a filename.
+     *
+     * @dataProvider providerTestFindReturnsCorrectString
+     */
+    public function testFindReturnsCorrectString($fileName, $expectedString)
+    {
+        $pathBuilder = new \FableProject\LaskGen\Core\PathBuilder($this->tree);
```
Comment (GSumisu) [inline src/test/TestPathBuilder.php]: changing namespace (comment above)

File: src/test/TestPathBuilder.php [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            array('', '')
+        );
+    }
+    
+    /**
+     * Standalone test to ensure invalid filenames are not found.
+     */
+    public function testFindDoesntFindInvalidFile()
+    {
+        $pathBuilder = new \FableProject\LaskGen\Core\PathBuilder($this->tree);
```
Comment (GSumisu) [inline src/test/TestPathBuilder.php]: changing namespace (comment above)

File: src/test/TestPathBuilder.php [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    }
+    
+    /**
+     * Test that the make method returns the correct path, including filename.
+     *
+     * @dataProvider providerTestMakeReturnsFullString
+     */
+    public function testMakeReturnsFullString($fileName, $expectedString)
+    {
+        $pathBuilder = new \FableProject\LaskGen\Core\PathBuilder($this->tree);
```
Comment (GSumisu) [inline src/test/TestPathBuilder.php]: changing namespace (comment above)

**Pull request merged**